README.md

```diff
@@ -172,6 +173,19 @@ class Object {
 (for instance, within closures) much more apparent and will make you think more on the reasons why you are using it.
 In closures, think about: should `self` be `weak` instead of `strong`? Apple has even rejected a request to enforce use of `self` for this reason, [among others](http://ericasadun.com/2016/01/06/the-swift-evolution-proposal-se-0009-rejection/).
 
+### Structs & Classes ###
+
+In Swift, structs maintain value semantics which means their values are copied when assigned. Classes, on the other hand, act like pointers from C
+and Objective-C; they are called reference types and the internal data is shared amongst instances of assigning.
+
+When composing your types, consider carefully what they're going to be used for before choosing what they should end up being. In general, 
+types that are immutable, stateless, and can have a definition for equality should be `struct`; for cases where an object is stateful, mutable,
+or where equality does not make sense, favor `class`.
+
+In general, you should favor structs and protocols over classes; even in cases where polymorphism would dictate the usage of a class, consider if you can
+achieve a similar result via protocols and extensions.
```

Comment:
if we could mention the keyword `composition` versus `inheritance` it would make me happy ;)